Exiled.API/Extensions/LiftTypeExtension.cs

```diff
@@ -0,0 +1,99 @@
+// -----------------------------------------------------------------------
+// <copyright file="LiftTypeExtension.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Extensions
+{
+    using System.Collections.Generic;
+
+    using Exiled.API.Enums;
+    using Exiled.API.Features;
+
+    /// <summary>
+    /// Contains an extension method to get <see cref="ElevatorType"/> from <see cref="Lift"/>.
+    /// Internal class <see cref="RegisterElevatorTypesOnLevelLoad"/> to cache the <see cref="ElevatorType"/> on level load.
+    /// </summary>
+    public static class LiftTypeExtension
+    {
+        private static readonly Dictionary<int, ElevatorType> OrderedElevatorTypes = new Dictionary<int, ElevatorType>();
+
+        /// <summary>
+        /// Gets the <see cref="ElevatorType"/>.
+        /// </summary>
+        /// <param name="lift">The <see cref="Lift"/> to check.</param>
+        /// <returns>The <see cref="ElevatorType"/>.</returns>
+        public static ElevatorType Type(this Lift lift) => OrderedElevatorTypes.TryGetValue(lift.GetInstanceID(), out var elevatorType) ? elevatorType : ElevatorType.Unknown;
+
+        /// <summary>
+        /// Gets all the <see cref="ElevatorType"/> values for for the <see cref="Lift"/> instances using <see cref="Lift.elevatorName"/> and <see cref="UnityEngine.GameObject"/> name.
+        /// </summary>
+        internal static void RegisterElevatorTypesOnLevelLoad()
```

Comment:
Call that method from the `Map::Lifts` getter, the same implementation as with the doortype exception